src/com/zs/exercise1/Bodmas.java

```diff
@@ -0,0 +1,140 @@
+            case '^' : return power(a,b);
+        }
+        return 0;
+    }
+
+    // This function solve the bodmas expression
+    public void bodmasSolver(String exp) {
+        int length = exp.length();
+        int ans=0;
+        Stack<Integer> value = new Stack<Integer>();
+        Stack<Character> operator = new Stack<Character>();
+
+        // use try-catch in case of any exception occur
+        try {
+            for (int i = 0; i < length; i++) {
+                if (exp.charAt(i) == ' ') {
+                    continue;
+                } else if (exp.charAt(i) == '(') {
+                    operator.push(exp.charAt(i));
+                } else if (isDigit(exp.charAt(i))) {            // when it is numeric value
+                    int val = 0;
+                    while (i < length && isDigit(exp.charAt(i))) {
+                        val = val * 10 + (exp.charAt(i) - '0');
+                        i++;
+                    }
+                    i--;
+                    value.push(val);
+
+                } else if (exp.charAt(i) == ')') {              // when the bracket expression will complete
+                    while (operator.peek() != ')') {
+                        int value2 = value.peek();
+                        value.pop();
+
+                        int value1 = value.peek();
+                        value.pop();
+
+                        char op = operator.peek();
+                        operator.pop();
+
+                        int output = perform(value1, value2, op);
+                        value.push(output);
+                    }
+                    operator.pop();
+                } else {
+                    // solve the arthmatic expression
+                    while ( !operator.empty() && precedence(operator.peek()) >= precedence(exp.charAt(i))) {
+                        int value2 = value.peek();
+                        value.pop();
+
+                        int value1 = value.peek();
+                        value.pop();
+
+                        char op = operator.peek();
+                        operator.pop();
+
+                        int output = perform(value1, value2, op);
+                        value.push(output);
+                    }
+                    operator.push(exp.charAt(i));
+                }
+            }
+
+            // in case of any operator remain in stack
+            while ( !operator.empty()) {
+                int value2 = value.peek();
+                value.pop();
+
+                int value1 = value.peek();
+                value.pop();
+
+                char op = operator.peek();
+                operator.pop();
+
+                int output = perform(value1, value2, op);
+                value.push(output);
+            }
+        }
+        catch(Exception ex) {
+            System.out.println("You give wrong expression ");
+            return ;
+        }
+        System.out.println(value.peek() );
+    }
+}
```

Comment:
function bodmasSolver is not being referenced anywhere.
Also your program is giving "You give wrong expression " for (5+8)*2 please check.